draft-ietf-bidding-and-auction-services.md

```diff
@@ -978,15 +992,40 @@ response from Bidding and Auction Services. It takes as input the
      parsed as a [URL], returning failure if there is an error.
 1. If `response["adMetadata"]` exists and is a string set
    `processed response["ad metadata"]` to `response["adMetadata"]`.
-1. If `response["buyerReportingId"]` exists and is a string, set
+1. If `response["buyerReportingId"]` exists and is a string set
    `processed response["buyer reporting id"]` to `response["buyerReportingId"]`.
-1. If `response["buyerAndSellerReportingId"]` exists and is a string, set
+1. If `response["buyerAndSellerReportingId"]` exists and is a string set
    `processed response["buyer and seller reporting id"]` to
    `response["buyerAndSellerReportingId"]`.
-1. If `response["selectedBuyerAndSellerReportingId"]` exists and is a string, set
-   `processed response["selected buyer and seller reporting id"]` to
-   `response["selectedBuyerAndSellerReportingId"]`.
-
+1. If `response["debugReports"]` exists and is an array:
+   1. For each `per origin debug reports` in `response["debugReports"]`:
+      1. If `per origin debug reports["adTechOrigin"]` does not exist or is not a string,
+         continue with the next iteration.
+      1. Let `ad tech origin` be `per origin debug reports["adTechOrigin"]` parsed as an [ORIGIN],
+         continue with the next iteration if there is an error.
+      1. If `per origin debug reports["reports"]` does not exist or is not an array,
+         continue with the next iteration.
+      1. For each `report` in `per origin debug reports["reports"]`:
+         1. If `report` is not a map, continue with the next iteration.
+         1. Let `component win` be `report["componentWin"]` if it exists and is a bool,
+            otherwise false.
+         1. If `report["url"]` exists and is a string:
+            1. Let `url` be `report["url"]` parsed as a [URL], or continue with the next

```

Comment:
should we do https check in the IETF spec?

Reply:
I think it makes more sense to check for a "possibly trustworthy" URL in the W3C spec.

Reply:
ack. Thanks